src/Symfony/Component/ObjectMapper/ObjectMapper.php

```diff
@@ -0,0 +1,226 @@
+                $map = $attr->newInstance();
+                $if = $map->if;
+
+                if (false === $if) {
+                    continue 2;
+                }
+
+                if ($if && ($fn = $this->getCallable($if)) && !$this->call($fn, null, $object)) {
+                    continue 2;
+                }
+
+                break;
+            }
+
+            $mapToProperty = $map?->to ?? $propertyName;
+            if (!$mapToProperty || !$toRefl->hasProperty($mapToProperty)) {
+                continue;
+            }
+
+            $value = $this->propertyAccessor ? $this->propertyAccessor->getValue($object, $propertyName) : $object->{$propertyName};
+            if ($map && $map->transform) {
+                $value = $this->applyTransforms($map, $value, $object);
+            }
+
+            if (
+                \is_object($value)
+                && ($innerMetadata = $this->metadataFactory->create($value))
+                && ($mapTo = $this->getMapTo($innerMetadata, $value, $object))
+                && (\is_string($mapTo->to) && class_exists($mapTo->to))
+            ) {
+                $value = $this->applyTransforms($mapTo, $value, $object);
+
+                if ($value === $object) {
+                    $value = $mapped;
+                } elseif ($objectMap->contains($value)) {
+                    $value = $objectMap[$value];
+                } else {
+                    $value = $this->map($value, $mapTo->to);
+                }
+            }
+
+            if (\array_key_exists($mapToProperty, $arguments)) {
+                $arguments[$mapToProperty] = $value;
+            } else {
+                $this->propertyAccessor ? $this->propertyAccessor->setValue($mapped, $mapToProperty, $value) : ($mapped->{$mapToProperty} = $value);
+            }
+        }
+
+        if ($arguments) {
+            $constructor?->invokeArgs($mapped, $arguments);
+        }
+
+        if ($objectMapInitialized) {
+            $objectMap = null;
+        }
+
+        return $mapped;
+    }
+
+    /**
+     * @param callable(): mixed $fn
+     */
+    private function call(callable $fn, mixed $value, object $object): mixed
+    {
+        $refl = new \ReflectionFunction(\Closure::fromCallable($fn));
+        $withParameters = $refl->getParameters();
+        $withArgs = [$value];
+
+        // Let's not send object if we don't need to, gives the ability to call native functions
+        foreach ($withParameters as $parameter) {
+            if ('object' === $parameter->getName()) {
+                $withArgs['object'] = $object;
+                break;
+            }
+        }
+
+        return \call_user_func_array($fn, $withArgs);
+    }
+
+    private function getCallable(string|callable $fn = null): callable|string|null
+    {
+        if (!$fn || !\is_string($fn)) {
+            return $fn;
+        }
+
+        if ($this->serviceLocator && $this->serviceLocator->has($fn)) {
```

Comment:
I'm interested in seeing how this locator will be populated (shouldn't the vairable be named $container BTW?)
Some basic FWB integration might be nice (having a tag to collect all these callables, and an attribute to autoconfigure them?)

Reply:
:+1: I think this should be done like the routing component

Reply:
:+1: for removing this ATM it could be done in a new PR, this will make it simpler for a start